### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by JantuDeb**
# Profile
- user name, email, photo
- update name, email & photo
- logout functionality

image
image

### CODE REVIEW

File: src/components/VideoNote.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import React, { useState } from "react";
+import { BiEdit } from "react-icons/bi";
+import { MdDelete } from "react-icons/md";
+import { useVideos } from "../context/videos/VideoContext";
+
+const VideoNote = ({ note, videoId }) => {
+  const [text, setText] = useState(note);
+  const [editing, setEditing] = useState(false);
+  const { addNote, deleteNote } = useVideos();
+  console.log(note);
```
Comment (rkapoor10) [inline src/components/VideoNote.jsx:10]: remove if not required

File: src/context/videos/VideoContext.jsx [hunk 2]
```diff
@@ -1,20 +1,111 @@
...
+  /**
+   * GET VIDEOS, CATEGORY FORM API
+   */
+  const getVideos = async () => {
+    try {
       const [videoResponse, categoryResponse] = await Promise.all([
         axiosInstance.get("/videos"),
         axiosInstance.get("/categories"),
       ]);
-      setVideos(videoResponse.data.videos);
```
Comment (rkapoor10) [inline src/context/videos/VideoContext.jsx:15]: Liked this approach

File: src/pages/profile/Profile.jsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  const [imgURL, setImgURL] = useState(false);
+  const [edinting, setEditing] = useState(false);
+
+  const { email, name } = userData;
+  const inputChangeHandler = (e) => {
+    setUserData((user) => ({ ...user, [e.target.name]: e.target.value }));
+  };
+
+  const editProfilePhoto = () => {
+    console.log(user.photo?.secure_url);
```
Comment (rkapoor10) [inline src/pages/profile/Profile.jsx:30]: remove console log if not needed

File: src/pages/profile/Profile.jsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    setImgURL(user.photo?.secure_url);
+    setShowModal(true);
+  };
+
+  const fileChangeHandler = (e) => {
+    console.log(e.target.files[0]);
```
Comment (rkapoor10) [inline src/pages/profile/Profile.jsx:36]: here also

**Review by rkapoor10 — APPROVED** [review]
The code is well written and I liked the edit profile option as well. Overall the code is good to merge just clean up the console logs

**Review by sushil-mod — APPROVED** [review]
Profile page working good . I liked the feature that we can change avatar image as well that looks  awesome . overall good code

**Pull request merged**